pkg/component/controller/extensions_controller.go

```diff
@@ -233,24 +235,34 @@ func (cr *ChartReconciler) updateOrInstallChart(ctx context.Context, chart v1bet
 			return fmt.Errorf("can't reconcile installation for `%s`: %v", chart.GetName(), err)
 		}
 	} else {
-		// update
-		chartRelease, err = cr.helm.UpgradeChart(chart.Spec.ChartName,
-			chart.Status.Version,
-			chart.Status.ReleaseName,
-			chart.Status.Namespace,
-			chart.Spec.YamlValues(),
-			timeout,
-		)
-		if err != nil {
-			return fmt.Errorf("can't reconcile upgrade for `%s`: %v", chart.GetName(), err)
+		if cr.chartNeedsUpgrade(chart) {
+			// update
+			chartRelease, err = cr.helm.UpgradeChart(chart.Spec.ChartName,
+				chart.Status.Version,
+				chart.Status.ReleaseName,
+				chart.Status.Namespace,
+				chart.Spec.YamlValues(),
+				timeout,
+			)
+			if err != nil {
+				return fmt.Errorf("can't reconcile upgrade for `%s`: %v", chart.GetName(), err)
+			}
 		}
 	}
 	cr.updateStatus(ctx, chart, chartRelease, nil)
 	return nil
 }
 
+func (cr *ChartReconciler) chartNeedsUpgrade(chart v1beta1.Chart) bool {
+	return !(chart.Status.Namespace == chart.Spec.Namespace &&
+		chart.Status.ReleaseName == chart.Spec.ReleaseName &&
+		chart.Status.Version == chart.Spec.Version &&
+		chart.Status.ValuesHash == chart.Spec.HashValues())
+}
+
```

Comment:
Predicates are the first things to break. It would be worth having a test for this especially since this now controls if the upgrade happens.

Reply:
Yeah, added the test. Thanks!